pulpcore/app/models/base.py

```diff
@@ -78,8 +78,13 @@ def save(self, *args, **kwargs):
         # on instances of MasterModel by the string stored in that model's TYPE attr.
         # Storing this type in a column on the MasterModel next to makes it trivial
         # to filter for specific detail model types across master's relations.
-        if not self.type:
-            self.type = self.TYPE
+        # Always prepend the type defined on a detail model with a django app label.
+        if self.type:
+            detail_model_type = self.type
+        else:
+            detail_model_type = self.TYPE
+        self.type = '{app_label}.{type}'.format(app_label=self._meta.app_label,
+                                                type=detail_model_type)
```

Comment:
@dralley , I agree with your concern, thanks for pointing out. I'll update the PR.

Yeah, initially it was the way you wrote, it just means that we can't guarantee proper namespacing for types. If plugin sets a type field themselves, we are not doing anything, it can be in any format possible.
The benefit is that it's more flexible for plugin writers, though I'm not sure how critical it is for them to control the type field fully.
The downside is that we would never be able to rely on the type field to identify which plugin it came from, if we ever need to write a migration for the table related to the master model.

Reply:
I think that's an acceptable risk. The plugins shouldn't set the type field themselves unless they know what they are doing and even then they are on their own if they do so.